sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -818,7 +818,8 @@ class Analyzer(
           failAnalysis(s"$expr has multiple Window Specifications ($distinctWindowSpec)." +
             s"Please file a bug report with this error message, stack trace, and the query.")
         } else {
-          distinctWindowSpec.head
+          val unbounded = SpecifiedWindowFrame(RowFrame, UnboundedPreceding, UnboundedFollowing)
```

Comment:
I think extracting the grouping by and order by clauses is better, and it also conveys the message that any Window operator should be capable of processing multiple frames. The impact of the PR will be a bit larger in that case; I'll have a look at it tomorrow.